generators/generator-constants.js

```diff
@@ -26,7 +26,7 @@ export const JHIPSTER_DEPENDENCIES_VERSION = '8.7.1-SNAPSHOT';
 // Version of Java
 export const JAVA_VERSION = '17';
 // Supported Java versions, https://www.oracle.com/java/technologies/java-se-support-roadmap.html
-export const JAVA_COMPATIBLE_VERSIONS = ['17', '21', '22'];
+export const JAVA_COMPATIBLE_VERSIONS = ['17', '21', '22', '23'];
```

Comment:
Java 22 should be dropped.
```suggestion
export const JAVA_COMPATIBLE_VERSIONS = ['17', '21', '23'];
```

Reply:
Java 22 can still be used with JHipster. Why should we prevent people from using it?

Reply:
It can be used by customizing build tool, but I think we should keep out-of-the-box compatibility with supported java versions https://www.oracle.com/java/technologies/java-se-support-roadmap.html.

Reply:
I get what you're saying, but that's just Oracle's support for their JDK distribution. Other JDK vendors might provide LTS support for different versions.

Reply:
Adoptium? https://adoptium.net/support
I think we should take a jvm support schedule.

Reply:
It looks like Azul is following a similar schedule: https://www.azul.com/products/azul-support-roadmap/
Microsoft too: https://learn.microsoft.com/en-us/java/openjdk/support